test/extended/builds/pipeline.go

```diff
@@ -64,15 +64,16 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 		mavenSlavePipelinePath        = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "maven-pipeline.yaml")
 		mavenSlaveGradlePipelinePath  = exutil.FixturePath("testdata", "builds", "gradle-pipeline.yaml")
 		//orchestrationPipelinePath = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "mapsapp-pipeline.yaml")
-		blueGreenPipelinePath         = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "bluegreen-pipeline.yaml")
-		clientPluginPipelinePath      = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "openshift-client-plugin-pipeline.yaml")
-		envVarsPipelinePath           = exutil.FixturePath("testdata", "samplepipeline-withenvs.yaml")
-		origPipelinePath              = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "samplepipeline.yaml")
-		configMapPodTemplatePath      = exutil.FixturePath("testdata", "config-map-jenkins-slave-pods.yaml")
-		imagestreamPodTemplatePath    = exutil.FixturePath("testdata", "imagestream-jenkins-slave-pods.yaml")
-		imagestreamtagPodTemplatePath = exutil.FixturePath("testdata", "imagestreamtag-jenkins-slave-pods.yaml")
-		podTemplateSlavePipelinePath  = exutil.FixturePath("testdata", "jenkins-slave-template.yaml")
-		secretPath                    = exutil.FixturePath("testdata", "openshift-secret-to-jenkins-credential.yaml")
+		blueGreenPipelinePath                  = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "bluegreen-pipeline.yaml")
+		clientPluginPipelinePath               = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "openshift-client-plugin-pipeline.yaml")
+		envVarsPipelinePath                    = exutil.FixturePath("testdata", "samplepipeline-withenvs.yaml")
+		origPipelinePath                       = exutil.FixturePath("..", "..", "examples", "jenkins", "pipeline", "samplepipeline.yaml")
+		configMapPodTemplatePath               = exutil.FixturePath("testdata", "config-map-jenkins-slave-pods.yaml")
+		imagestreamPodTemplatePath             = exutil.FixturePath("testdata", "imagestream-jenkins-slave-pods.yaml")
+		imagestreamtagPodTemplatePath          = exutil.FixturePath("testdata", "imagestreamtag-jenkins-slave-pods.yaml")
+		podTemplateSlavePipelinePath           = exutil.FixturePath("testdata", "jenkins-slave-template.yaml")
+		multiNamespaceClientPluginPipelinePath = exutil.FixturePath("testdata", "multi-namespace-pipeline.yaml")
```

Comment:
since every one of these tests has to stand up a jenkins instance and do stuff, can we combine a few of these into a single pipeline?  Or at least run them using the same jenkins instance? Every one of these tests is like 5+ minutes, right? (due to waiting for jenkins to start) Even parallelism only gets us so far and if we have any hope of making the builds extended test be a blocking job we have to keep it short.

Reply:
Yeah, this is a long time friction point.  You've got
a) after many permutations, the only effective way to start jenkins only once was to use BeforeSuite, which would do it for all of core.sh ... all the other attempts resulted in a jenkins getting started before each g.It or g.Context ... and no, the jenkins_plugin.go attempt of creating it in a separate project did not help

b) coupled with a) ... there being no seeming way to override the coupling of namespace with each Describe/Context/It

c) getting smaller granularity of tests, such that a failure of one in the beginning / middle does not prevent additional tests from getting executed

d) the now much higher mem/cpu demand on startup with jenkins since v2.89

I've got a commit I'm about to push (making final clean up now) that finally capitulates to d) and sacrifices some of c)

It reduces the 12 or so pipeline.go tests into 3 ... the client plugin tests are separated out since our pr testing for that plugin needs to leverage a ginkgo focus that executes just those ... the other 2 delineate the new use of the persistent template @coreydaley introduced

Reply:
yeah i'm willing to give up some of (c), as long as we don't have major flakiness, stopping on first failure seems ok (does make iterating through fixing tests a bit harder, admittedly).

similarly i'm ok w/ us adding memory/cpu when we stand up jenkins for our tests, as long as somewhere we're making sure it still runs in a 512meg container so we know we haven't completely broken starter.

Reply:
(and yes i agree long term we need to solve the starter problem differently)